src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-
-        """
-        return candidate_matches[
-            candidate_matches.record_id_eia_deprish ==
-            candidate_matches.record_id_eia_ferc1]
-
-    def get_matches_at_diff_ownership(self, candidate_matches):
-        """
-        Get matches when the record_id_eia matches except for the ownership.
-
-        The master unit list includes various levels of ownership associated
-        with each record. Some are labeled "owned" (for the utilty's owned
-        portion of the plant part) and some are labeled "total" (for the full
-        plant part).
-
-        The method selects the matches where the potential matches have the
-        same EIA record id expect for the ownership level (owned vs total).
-
-        Note: Is there a cleaner way to do this??
+    @property
+    def allocator_cols_dict(self) -> Dict:
+        """Grab the columns from the metadata which need to be allocated."""
+        return {
+            allocate_col: self.treatments[allocate_col].allocator_cols
+            for allocate_col in self.get_treatment_cols('scale')
+        }
 
-        Args:
-            candidate_matches (pandas.DataFrame): dataframe of the matches
-                of possible matches between the ferc1 and deprecation.
-        """
-        diff_own = (
-            candidate_matches[
-                candidate_matches.record_id_eia_deprish.str.replace(
-                    '(owned|total)', "", regex=True) ==
-                candidate_matches.record_id_eia_ferc1.str.replace(
-                    '(owned|total)', "", regex=True)
-            ]
+    @property
+    def plant_part_pk_cols(self):
+        """Get the primary keys for a plant-part."""
+        return (pudl.analysis.plant_parts_eia.PLANT_PARTS
+                [self.plant_part]['id_cols']
+                + pudl.analysis.plant_parts_eia.IDX_TO_ADD
+                + pudl.analysis.plant_parts_eia.IDX_OWN_TO_ADD)
+
+    def execute(self, df_to_scale: pd.DataFrame, ppl: pd.DataFrame):
+        """
+        Scale a dataframe to a generator-level.
+
+        There are four main steps here:
+        * STEP 1: Aggregate the dataset records that are associated with the
+          same EIA plant-part records. Sometime the original dataset include
+          mulitple records that are associated with the same plant-part record.
+          We are assuming that the duplicate records are not reporting
+          overlapping data (i.e. one depreciation record for a unit and another
+          for the emissions control costs associated with the same unit). We
+          know this isn't always a perfect bet.
+        * STEP 2: Merge in the EIA plant-part list generators. This is
+          generally a one-to-many merge where we cast many generators across
+          each data-set record. Now we have the dataset records associated with
+          generators but duplicated. (Step 2 and 3 are grouped into one method)
+        * STEP 3: This is the splitting/scaling step. Here we take the
```

Comment:
"splitting/scaling" = just scaling? Or allocating/aggregating?

Reply:
good catch! I thought i searched for and took out all instances of the old names.

Reply:
I've now removed all SPLITS